pkg/services/blockfetcher/blockfetcher.go

```diff
@@ -441,7 +459,7 @@ func (bfs *Service) exiter() {
 	bfs.log.Info("shutting down NeoFS BlockFetcher service",
 		zap.Bool("force", force),
 	)
-
+	bfs.isActive.CompareAndSwap(true, false)
```

Comment:
Move it here for retry and also https://github.com/nspcc-dev/neo-go/issues/3699 solving with it. @AnnaShaleva is it a possible solution?